### PR TITLE
Remove brackets from comments, strings and regexes before evaluating the indentation

### LINKED ISSUE #209862: Explore removing brackets of language configuration file from the string/regex before evaluating indentation
In relation to https://github.com/microsoft/vscode/issues/209519

In the case of a single line string, or a string that appears in conjunction with other tokens on a line, we would like to remove the brackets of the language configuration file from the string, before evaluating the indentation rules on that stripped string. We want to do this to avoid indenting in cases such as the following:

`const aString = 'function f() {'`

Do the same for brackets inside of the regexes.

### PR DESCRIPTION
**Pull request by aiday-mar**
fixes https://github.com/microsoft/vscode/issues/209862

I have made an attempt to combine the methods of the `indentation.test.ts` and `autoindent.test.ts` files, but this caused the following error: `uncaught ReferenceError: window is not defined`. I suppose this is because the `autoindent.test.ts` file contains node.js tests which were accessing methods from the non node.js tests in `indentation.test.ts`.

### CODE REVIEW

File: src/vs/editor/common/languages/enterAction.ts [hunk 1]
```diff
@@ -7,43 +7,25 @@ import { Range } from 'vs/editor/common/core/range';
...
 
 export function getEnterAction(
 	autoIndent: EditorAutoIndentStrategy,
 	model: ITextModel,
 	range: Range,
 	languageConfigurationService: ILanguageConfigurationService
 ): CompleteEnterAction | null {
-	const scopedLineTokens = getScopedLineTokens(model, range.startLineNumber, range.startColumn);
-	const richEditSupport = languageConfigurationService.getLanguageConfiguration(scopedLineTokens.languageId);
+	model.tokenization.forceTokenization(range.startLineNumber);
```
Comment (bhavyaus) [inline src/vs/editor/common/languages/enterAction.ts:19]: Test comment

Comment (bhavyaus) [reply src/vs/editor/common/languages/enterAction.ts:19]: Test comment

File: src/vs/editor/common/languages/languageConfigurationRegistry.ts [hunk 2]
```diff
@@ -9,7 +9,6 @@ import * as strings from 'vs/base/common/strings';
 import { ITextModel } from 'vs/editor/common/model';
 import { DEFAULT_WORD_REGEXP, ensureValidWordDefinition } from 'vs/editor/common/core/wordHelper';
 import { EnterAction, FoldingRules, IAutoClosingPair, IndentationRule, LanguageConfiguration, AutoClosingPairs, CharacterPair, ExplicitLanguageConfiguration } from 'vs/editor/common/languages/languageConfiguration';
```
Comment (bhavyaus) [inline src/vs/editor/common/languages/languageConfigurationRegistry.ts:11]: test

Comment (bhavyaus) [reply src/vs/editor/common/languages/languageConfigurationRegistry.ts:11]: test

**Review by hediet — APPROVED** [review]

**Pull request merged**